pkg/arvo/gen/story-read.hoon

```diff
@@ -0,0 +1,122 @@
+::::
+++  story-read
+  |=  [[our=ship syd=^desk cas=case] current-commit=yaki:clay tale=story]
+  ^-  tang
+  %-  head  :: result from state
+  =|  state=[result=tang mergebase=(unit tako:clay)]
+  |-
+  ^-  _state
+  =*  commit-loop  $
+  =/  reverse-ancestors  (flop p.current-commit)
+  |-
+  =*  ancestor-loop  $
+  ?-    reverse-ancestors
+      ~
+    ::  stop here and return the current message
+    =/  msg=(list cord)  (msg-from-commit current-commit tale)
+    [(weld msg result.state) mergebase=~]
+  ::
+      [tako:clay ~]
+    =/  parent=tako:clay  i.reverse-ancestors
+    =/  parent-commit     .^(yaki:clay %cs /(scot %p our)/[syd]/(scot cas)/yaki/(scot %uv parent))
+    =/  msg               (msg-from-commit current-commit tale) 
+    ::  If there is a mergebase and we are visting it right now:
+    ::    stop here and clear the mergebase. 
+    ::    skip adding the mergebase's msg itself because it will be added through the other branch
+    ::  Otherwise, record the current message if exists
+    ?:  ?&(?=(^ mergebase.state) =(u.mergebase.state r.current-commit))
+      [result=result.state mergebase=~]
+    commit-loop(current-commit parent-commit, result.state (weld msg result.state))
+  ::
+      [tako:clay tako:clay ~]
+    ::
+    ::  mainline: ultimate base chain
+    ::  nowline: relative mainline
+    ::  sideline: side-chain, featurebranch
+    ::
+    ::  From the context of e, commit c is on its relative mainline, or nowline, while commit d is on its sideline
+    ::  %base  a--b-------------X  :: mainline
+    ::  %new      \--c------e--/   :: nowline
+    ::  %new2        \--d--/       :: sideline 
+    ::
+    ::  
+    =/  sideline=tako:clay               i.reverse-ancestors
+    =/  mainline=tako:clay               i.t.reverse-ancestors
+    =/  mergebases                       .^((list tako:clay) %cs /(scot %p our)/[syd]/(scot cas)/base-tako/(scot %uv mainline)/(scot %uv sideline))  :: XX base-tako ignores beak
+    =/  next-mergebase=(unit tako:clay)  ?~(mergebases ~ (some i.mergebases))  :: take the first valid mergebase (by convention) if exists, else none
+    :: ~&  "Debug: Mergebase is null? {<=(~ next-mergebase)>}"
+    =/  sideline-commit=yaki:clay        .^(yaki:clay %cs /(scot %p our)/[syd]/(scot cas)/yaki/(scot %uv sideline))
+    =/  mainline-commit=yaki:clay        .^(yaki:clay %cs /(scot %p our)/[syd]/(scot cas)/yaki/(scot %uv mainline))
+    =/  msg=(list cord)                  (msg-from-commit current-commit tale)
+    ::
+    ::  1 - process current commit
+    ::  2 - recur and queue processing on all commits on the sideline
+    ::  3 - recur and queue processing on all commits on the mainline
+    ::
+    ::  Because mainline messages are cons'd to result last, they are 
+    ::  (by definition) towards the less recent side of the resulting flopped list
+    ::
+    =.  state  [result=(weld msg result.state) mergebase=next-mergebase]  :: 1
```

Comment:
end comments should be aligned

Reply:
done in https://github.com/urbit/urbit/pull/5612/commits/a43dafdf4415297c800d66357d1b9546d108c1f7